catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/FeedRepository.java

```diff
@@ -181,6 +187,38 @@ public Thread addPostToThread(String id, Post post) throws IOException {
     return thread;
   }
 
+  public Optional<Post> getPostById(Thread thread, String postId) {
+    return thread.getPosts().stream().filter(p -> p.getId().equals(UUID.fromString(postId))).findAny();
+  }
+
+  @Transaction
+  public DeleteResponse<Post> deletePost(String threadId, String postId, String userName) throws IOException {
+    // validate the thread
+    Thread thread = get(threadId);
+    Post post;
+
+    // validate post
+    Optional<Post> optionalPost = getPostById(thread, postId);
+    if (optionalPost.isPresent()) {
+      // delete post only if the author tries to delete it
+      post = optionalPost.get();
+      if (userName.equals(post.getFrom())) {
+        List<Post> posts = thread.getPosts();
+        // Remove the post to be deleted from the posts list
+        posts = posts.stream().filter(p -> !p.getId().equals(UUID.fromString(postId))).collect(Collectors.toList());
+        thread.withUpdatedAt(System.currentTimeMillis()).withUpdatedBy(userName).withPosts(posts);
+        // update the json document
+        dao.feedDAO().update(threadId, JsonUtils.pojoToJson(thread));
+      } else {
+        // TODO Should we also allow admins to delete a post?
+        throw new AuthorizationException(CatalogExceptionMessage.noPermission(userName));
```

Comment:
Minor (for future): We may need to surface better error message than the one from here..

"X cannot delete post by user Y" or something

Reply:
Also allow admins to delete

Reply:
any Auth checks should be delegated to the authorizer that way its easier enforce and not have exceptions